components/form/CustomerForm.tsx

```diff
@@ -14,54 +14,36 @@ import {
     Box,
 } from '@chakra-ui/react';
 import React, { useState } from 'react';
-import { components } from '@/lib/types/api';
 import * as fetch from '@/lib/utils/fetch';
-import { FormStatus } from './reducer';
 import { useSWRConfig } from 'swr';
 import ErrorAlert from '@/components/common/ErrorAlert';
+import { FormStatus } from './ProjectForm';
+import { CustomerDTO } from '@/lib/types/dto';
+import { customerEndpointURL } from '@/lib/const';
 
-type StateType = {
```

Comment:
Usually there is no reason to prefer the use of single `useState` instead of multiple `useState` hooks where the values actually mean something. For example, in this case, it's much clearer to split this into a `CustomerDTO` state and separate states that are related to the form state.

Reply:
As a bonus, we reduce [clutter](https://github.com/three-consulting/epoc-frontend/pull/78/files#diff-0371e3443d5f2360efcc6f352cb9af86177163d97e5dcaf6fa718addc441eb26L63-L69)